### PR TITLE
ANDROID-11940 Add border radius attributes and migrate components

### PR DESCRIPTION
**Pull request by jeslat**
### :goal_net: What's the goal?
Add border radius attributes and migrate components.

This is the specification https://www.figma.com/file/PApssATUkePZnIOB4GmAeI/%F0%9F%94%B9-%E2%80%8E%E2%80%8ETokens-definition?node-id=1505-178&t=WUIk1fpiapliXMwS-0.

### :construction: How do we do it?
* In Compose implementation, new fields have been added to `MisticaValues` as we did with `titleStyle`.
* In XML, I've created a new styleable `BorderRadius` to provide the different radius attributes.

### ☑️ Checks
- [x] I updated the documentation, including readmes and wikis. If this is a breaking change, update [UPGRADING.md](../UPGRADING.md) to inform users how to proceed. If no updates are necessary, indicate so.
- [x] Tested with dark mode.
- [x] Tested with API 23.

### :test_tube: How can I test this?
- You can use the catalog generated in this PR.

### CODE REVIEW

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/CatalogMainActivity.kt [hunk 1]
```diff
@@ -90,7 +90,7 @@ class CatalogMainActivity : AppCompatActivity() {
             SectionItem("Lists", R.drawable.ic_lists, Section.LISTS),
             SectionItem("Load Error Feedback", R.drawable.ic_load_feedback_error, Section.LOAD_ERROR_FEEDBACK),
             SectionItem("Media Card", R.drawable.ic_cards, Section.MEDIA_CARDS),
-            SectionItem("Others", R.drawable.ic_others, Section.OTHERS),
+            SectionItem("Filters", R.drawable.ic_filters, Section.FILTERS),
```
Comment (jeslat) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/CatalogMainActivity.kt:93]: I've renamed this section to `Filters` which is the right name

**Review by dagonco — APPROVED** [review]
Great currazo! 👌🏻

File: library/src/main/java/com/telefonica/mistica/compose/theme/brand/Brand.kt [hunk 2]
```diff
@@ -19,8 +21,19 @@ interface Brand {
...
+            titleStyle = TitleStyle.TITLE_1,
+            buttonBorderRadius = 4.dp,
+            inputBorderRadius = 8.dp,
+            containerBorderRadius = 16.dp,
+            legacyDisplayBorderRadius = 16.dp,
+            popupBorderRadius = 8.dp,
+            checkboxBorderRadius = 2.dp,
+            indicatorBorderRadius = 999.dp,
+            sheetBorderRadius = 16.dp,
+            barBorderRadius = 999.dp
```
[the remark this answers is not in the document]
Comment (AnaMontes11) [reply library/src/main/java/com/telefonica/mistica/compose/theme/brand/Brand.kt]: > These are the default values

Don't you need to include avatarBorderRadius = 50%?

**Review by AnaMontes11 — APPROVED** [review]
Great job!

**Pull request merged**